def convert_to_int(value):
    """
    Функция пытается преобразовать входное значение в целое числою
    Обрабатывает исключения, если преобразование невозможно или возникает любое другое исключение.
    
      
    Аргументы:
    value: входное значение, которое нужно преобразовать в целое число.
          
    Возвращает:
    Целое число, если преобразование успешно.
    None, если произошло исключение.
    """
    try:
        # Попытка преобразования строки в целое число с помощью функции int()
        result = int(value)
        print(f"Успешное преобразование: {result}")
        return result
    except ValueError:
        # Обработка исключения ValueError, если строку невозможно преобразовать в целое число
        print("Ошибка: Невозможно преобразовать строку в целое число.")
    except Exception as e:
        # Обработка любого исключения
        print(f"Неожиданная ошибка: {e}")
    finally:
        # Блок finflly, который выполнится в любом случае
        print("Завершение попытки преобразования.") 

# Проверка функции с корректной строкой
convert_to_int("123")
# Прверка функции с некорректной строкой
convert_to_int("abc")
# Проверка функции с другим типом данных
convert_to_int([1,2,3])

